### PR TITLE
Add Github workflow to the Redis DB bal persist client

### PR DESCRIPTION
**Pull request by dinukaamarasinghe817**

### CODE REVIEW

File: build-config/spotbugs-exclude.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<FindBugsFilter>
+</FindBugsFilter>
```
Comment (Nuvindu) [inline build-config/spotbugs-exclude.xml:2]:
```suggestion
</FindBugsFilter>

```

File: gradle.properties [hunk 2]
```diff
@@ -0,0 +1,62 @@
+group=io.ballerina.stdlib
```
Comment (Nuvindu) [inline gradle.properties:1]:
```suggestion
group=io.ballerina.lib
```

Comment (dinukaamarasinghe817) [reply gradle.properties]: Fixed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/pull/3/commits/9a818a3bb5c329c06d928804881ca678c28e688f

File: build-config/spotbugs-exclude.xml (hunk 1 above)
Comment (dinukaamarasinghe817) [reply build-config/spotbugs-exclude.xml]: Fixed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/pull/3/commits/9a818a3bb5c329c06d928804881ca678c28e688f

**Review by daneshk — APPROVED** [review]
LGTM

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+plugins {
+    id "com.github.spotbugs" version "${githubSpotbugsVersion}"
+    id "com.github.johnrengelman.shadow" version "${githubJohnrengelmanShadowVersion}"
+    id "de.undercouch.download" version "${underCouchDownloadVersion}"
+    id "net.researchgate.release" version "${researchgateReleaseVersion}"
```
Comment (ThisaruGuruge) [inline build.gradle:23]: Shall we move the plugin management to the `settings.gradle` file? Check this out:

- [`settings.gradle` File](https://github.com/ballerina-platform/module-ballerina-graphql/blob/45577c19a65a456a134ca0a705466c0477f9c63c/settings.gradle#L10)
- [`build.gradle` File](https://github.com/ballerina-platform/module-ballerina-graphql/blob/45577c19a65a456a134ca0a705466c0477f9c63c/build.gradle#L19)

Notice that the plugin versions are handled in the settings.gradle file while the build.gradle files are just using those plugins, not handling the versions.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+description = 'Ballerina - Persist'
+
+def packageName = "persist.redis"
+
+ext.ballerinaLangVersion = project.ballerinaLangVersion
```
Comment (ThisaruGuruge) [inline build.gradle:30]: We don't need this ext definition.

File: gradle.properties [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+# Level 03
+stdlibCacheVersion=3.7.0
+stdlibMimeVersion=2.9.0
+stdlibUuidVersion=1.7.0
+
+# Level 04
+stdlibAuthVersion=2.10.0
+stdlibJwtVersion=2.10.0
+stdlibOAuth2Version=2.10.0
```
Comment (ThisaruGuruge) [inline gradle.properties:58]: When it comes to dependency management in the `gradle.properties`, it doesn't make much sense to differentiate the direct and transitive dependencies. IMO, it is better to just put them together by grouping them using the level.

File: gradle/libs.versions.toml
Comment (ThisaruGuruge) [inline gradle/libs.versions.toml:1]: We can remove this file.

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
Comment (ThisaruGuruge) [inline gradle/wrapper/gradle-wrapper.properties:3]: Let's use Gradle `8.2.1` version. Note that changing the version will require you to update the wrapper jar as well.

Comment (dinukaamarasinghe817) [reply gradle/wrapper/gradle-wrapper.properties]: All the suggestions have been addressed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/commit/500fc9b564e179cdc127ac649d628d404eb25608

File: .github/workflows/build-timestamped-master.yml
Comment (ThisaruGuruge) [inline .github/workflows/build-timestamped-master.yml:1]: Let's use the centralized workflow templates for these workflows.

**Pull request closed without merging**